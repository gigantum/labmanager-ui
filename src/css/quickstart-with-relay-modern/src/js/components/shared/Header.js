import React, { Component } from 'react'
import GigantumLogo from '../../../images/logos/gigantum.png';

export default class Header extends Component {
  render() {
    return (
      <div className={'header__container flex flex--row justify--space-between'}>
        <div className={'flex-1-0-auto'}>
          <h1 className='header__title text-center'>Gigantum</h1>
        </div>
        <div>
          <img alt='gigantum logo' src={GigantumLogo} height='60'/>
        </div>
      </div>
    )
  }
}
