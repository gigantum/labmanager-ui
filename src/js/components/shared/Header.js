import React, { Component } from 'react';
import { Link } from 'react-router-dom';
// components
import User from './User';


export default class Header extends Component {
  /**
    @param {}
    logout through Auth0
  */
  logout() {
    this.props.auth.logout();
  }
  render() {
    const { isAuthenticated } = this.props.auth;
    return (
      <div className="Header flex flex--row justify--space-between">
        <div className="flex justify--space-around flex-1-0-auto">

          <ul className="Header__nav flex flex--row justify--space-between">
            <li>
              <Link
                className="Header__nav-item Header__nav-item--datasets flex flex--row justify--space-between"
                to={{ pathname: '/datasets' }}
              >
                <div className="Header__datasets-icon" />
                Datasets
              </Link>
            </li>
            <li>
              <Link
                className="Header__nav-item Header__nav-item--labbooks flex flex--row justify--space-between"
                to={{ pathname: '/projects' }}
              >
                <div className="Header__labbook-icon" />
                Projects
              </Link>
            </li>

          </ul>

          {
            isAuthenticated() && (
            <User {...this.props} />
              )
          }
        </div>
      </div>
    );
  }
}
