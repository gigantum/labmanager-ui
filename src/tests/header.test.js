import React, { Component } from 'react'
import renderer from 'react-test-renderer';
import GigantumLogo from '../images/logos/gigantum.png';
import Header from './../js/components/shared/Header';

test('Header Reners Correctly', () => {
  const component = renderer.create(
    <Header />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
