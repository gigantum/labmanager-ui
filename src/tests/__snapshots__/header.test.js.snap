// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Reners Correctly 1`] = `
<div
  className="header__container flex flex--row justify--space-between"
>
  <div
    className="flex-1-0-auto"
  >
    <h3
      className="header__title text-center"
    >
      Gigantum
    </h3>
  </div>
</div>
`;

exports[`Test Header rendering 1`] = `
<div
  className="header__container flex flex--row justify--space-between"
>
  <div
    className="flex-1-0-auto"
  >
    <h3
      className="header__title text-center"
    >
      Gigantum
    </h3>
  </div>
</div>
`;

exports[`Test header rendering 1`] = `
<div
  className="header__container flex flex--row justify--space-between"
>
  <div
    className="flex-1-0-auto"
  >
    <h3
      className="header__title text-center"
    >
      Gigantum
    </h3>
  </div>
</div>
`;
